### PR TITLE
refactor(SetRegularKey): Update transaction to use new scheme

### PR DESCRIPTION
**Pull request by ckniffen**
### Context of Change

Updates SetRegularKey implementation to use the new transaction scheme that is documented [here](https://github.com/ripple/explorer/blob/staging/src/containers/shared/components/Transaction/README.md).

Add new tests for when transaction unsets the key and have assertions check the values shown to the user.

Part of #205 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

### CODE REVIEW

**Comment by JST5000** [comment]
Do you happen to know an account on mainnet or testnet with a regular key that I could use when looking at this locally to see how the changes affect the Regular Key view?

**Comment by ckniffen** [comment]
> Do you happen to know an account on mainnet or testnet with a regular key that I could use when looking at this locally to see how the changes affect the Regular Key view? @ckniffen

I used these in the unit tests.

### Set
https://livenet.xrpl.org/transactions/B10C4D6361B12B7F530ED6748293F869B50805BB66569577D2A4C0C1DF0F7E17

### Unset https://testnet.xrpl.org/transactions/1D25137B9384C89C7991C1EC702EAD285AB964F172B2D2AF9CBC51F63BD14AC4

**Review by JST5000 — APPROVED** [review]
LGTM :)

File: src/containers/shared/components/Transaction/SetRegularKey/parser.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export function parser(tx: any) {
+  return {
+    key: tx.RegularKey || null,
```
Comment (mvadari) [inline src/containers/shared/components/Transaction/SetRegularKey/parser.ts:3]: Isn't this already going to be undefined if `tx.RegularKey` doesn't exist?

Comment (ckniffen) [reply src/containers/shared/components/Transaction/SetRegularKey/parser.ts:3]: I can try that.  This was copied over from what was there before.

File: src/containers/Transactions/test/Description.test.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ import EscrowCancel from './mock_data/EscrowCancel.json'
 import PaymentChannelCreate from './mock_data/PaymentChannelCreate.json'
 import PaymentChannelClaim from './mock_data/PaymentChannelClaim.json'
 import PaymentChannelFund from './mock_data/PaymentChannelFund.json'
-import SetRegularKey from './mock_data/SetRegularKey.json'
+import SetRegularKey from '../../shared/components/Transaction/SetRegularKey/test/mock_data/SetRegularKey.json'
```
[the remark this answers is not in the document]
Comment (ckniffen) [reply src/containers/Transactions/test/Description.test.js:16]: This just happens to be the transaction that is used to test the entire Description tab so it is still needed.

**Comment by shawnxie999** [comment]
Just curious would be the purpose of transitioning to the new scheme? Having both schemes within the code right now takes a while to understand.

**Comment by shawnxie999** [comment]
nit: should we  add some comments for `getInstructions`,  `Simple`, `Description` and `TxDetails` where the new and old schemes are used? it wasnt exactly obvious as what they meant without knowing about this scheme transition

**Review by shawnxie999 — APPROVED** [review]

**Comment by ckniffen** [comment]
> nit: should we add some comments for `getInstructions`, `Simple`, `Description` and `TxDetails` where the new and old schemes are used? it wasnt exactly obvious as to what they meant without knowing about this scheme transition, like somewhere here

I will add that in my next PR I will probably have created tomorrow that covers SignerListSet.

**Pull request merged**